### PR TITLE
Set long_description in setup.py

### PR DESCRIPTION
**Pull request by c-bata**
Optuna doesn't provide the description at PyPI.
https://pypi.org/project/optuna/

*Caution: We can use markdown-style long_description from setuptools v38.6.0 because it started to support PEP566*
https://setuptools.readthedocs.io/en/latest/history.html#v38-6-0

### CODE REVIEW

**Comment by c-bata** [comment]
Hmm... I don't know why test-python35 is failed.

**Comment by sile** [comment]
### FYI

> Hmm... I don't know why test-python35 is failed.

I don't know the exact reason of the failure too, but in my environment (python-3.5.3), the following diff fixed the problem.
```diff
diff --git a/setup.py b/setup.py
index 2e058c6..152c5c1 100644
--- a/setup.py
+++ b/setup.py
@@ -100,6 +100,7 @@ setup(
             'storages/rdb/alembic/versions/*.*'
         ]
     },
+    data_files=[('.', ['./README.md'])],
     install_requires=get_install_requires(),
     tests_require=get_extras_require()['testing'],
     extras_require=get_extras_require(),
```

**Comment by c-bata** [comment]
Thank you! I'll check later.

BTW, I notice that Optuna is in PyPI trending projects now. It looks a lot of users download optuna today.
https://pypistats.org/packages/optuna

スクリーンショット 2019-06-07 19 03 01

Congratulation!

**Comment by sile** [comment]
Great news🎉 
Thanks for letting us know!

**Comment by c-bata** [comment]
Hmm... Although I applied your patch, `tests-python35` is failed. Basically, It doesn't need to specify README.md at `data_files` because setuptools include it at following lines:
https://github.com/pypa/setuptools/blob/8aeff6b2c9a64e47ad2a22533d7e65c08cd4103f/setuptools/command/sdist.py#L40-L41

I couldn't reproduce this error in my macOS environment even if I install optuna in python3.5 without specifying `data_files` argument of `setup` function:

```
$ python3.5 -m venv venv35
$ source venv35/bin/activate
$ python setup.py sdist
$ pip install ./dist/optuna-0.12.0.tar.gz
...
Successfully installed optuna-0.12.0
```

I'll investigate the reason of this error later by using login with ssh feature of circleci.

**Comment by sile** [comment]
> Although I applied your patch, `tests-python35` is failed

Hmm...

**Comment by c-bata** [comment]
Sorry, I couldn't find the reason why README is not found on Python3.5. Provisionally, I added an exception handling. Please close this PR if it needs to build `sdist` package in Python3.5.

**Comment by c-bata** [comment]
I noticed another problem to render Optuna's logo image. It feels better to close this.

**Comment by toshihikoyanase** [comment]
> the reason why README is not found on Python3.5.

Maybe the cause is the combination of old setuptools and the CircleCI environment. When I add an update command of `setuptools` to the `install` step of CircleCI, I can successfully build the sdist package using Python 3.5.

Please take a look at the following branch:
https://github.com/pfnet/optuna/compare/set-long-description-repro

Changes
- Add update command of [`setuptools`](https://github.com/pfnet/optuna/compare/set-long-description-repro#diff-1d37e48f9ceff6d8030570cd36286a61R106)
- Revert [`get_long_description`](https://github.com/pfnet/optuna/compare/set-long-description-repro#diff-2eeaed663bd0d25b7e608891384b7298R31-R37)

> I noticed another problem to render Optuna's logo image at pypi.org because this specifies relative path.

Thank you for pointing it out. I think we can use the absolute path as follows:

```
<div align="center"><img src="https://raw.githubusercontent.com/pfnet/optuna/master/docs/image/optuna-logo.png" width="800"/></div>
```

**Comment by c-bata** [comment]
> Maybe the cause is the combination of old setuptools and the CircleCI environment. When I add an update command of setuptools to the install step of CircleCI, I can successfully build the sdist package using Python 3.5.

Oh, I see! Thank you @toshihikoyanase !

It looks your changes good. I'll cherry-pick your changes and specify absolute path.

**Comment by c-bata** [comment]
@toshihikoyanase I rebased on your branch and updated a link URL of the logo image.

File: setup.py [hunk 1]
```diff
@@ -27,6 +28,13 @@ def get_version():
     assert False
 
 
+def get_long_description():
+    # type: () -> str
```
Comment (toshihikoyanase) [inline setup.py]: Please add a blank line after the type hint.
I'm sorry that this is my mistake originally.

**Comment by c-bata** [comment]
It's OK! I pushed 6300ce71897c28ff23dc6e196499f08224496fe0 👍

**Pull request merged**